src/Common/getMaxFileDescriptorCount.cpp

```diff
@@ -8,29 +8,32 @@
 int getMaxFileDescriptorCount()
 {
     namespace fs = std::filesystem;
-    int result = -1;
-#if defined(OS_LINUX) || defined(OS_DARWIN)
-    using namespace DB;
-
-    if (fs::exists("/proc/sys/fs/file-max"))
-    {
-        ReadBufferFromFile reader("/proc/sys/fs/file-max");
-        readIntText(result, reader);
-    }
-    else
+    static int result = []
     {
-        auto command = ShellCommand::execute("ulimit -n");
-        try
+        int ret = -1;
+    #if defined(OS_LINUX) || defined(OS_DARWIN)
+        using namespace DB;
+
+        if (fs::exists("/proc/sys/fs/file-max"))
```

Comment:
@chhetripradeep okay now it will be called once whatever we do
what do you think would be the best way to fetch this info

also, it seems getrlimit returns invalid value in macOS not sure if the same is true for `ulimit -n`

Reply:
getrlimit() is part of POSIX standards and macOS being posix compliant should have this functionality. Google returns some [doc](https://developer.apple.com/library/archive/documentation/System/Conceptual/ManPages_iPhoneOS/man2/getrlimit.2.html)

I tried it on my machine: https://pastila.nl/?016bc976/7a16d656958d9dafc21552ebc930d3dc#sA89nR102rm4eRReMg+oGw==

Reply:
Now that we are running it only, I think we can continue using `ulimit -n`for macos.

Reply:
> I know that it works on macOS but it doesn't mean it will return correct value.

I don't think this is a good reason to not use it. Reading `proc/sys/fs/file-max` or `ShellCommand::execute` is much worse than a simple syscall via `getrlimit`.